### PR TITLE
Fix style & genericness

### PR DESCRIPTION
**Pull request by bb010g**
Probably what will end up being early stages right now, but a PR to just generally clean up aura.
* [x] `map` → `fmap` (and `<$>` where appropriate)
* [x] `return` → `pure`
* [x] `>>` → `*>` (and `$>` where appropriate)
* [x] `(a,b)` → `(a, b)`
* [x] List funs → Foldable/Traversable funs
* [x] Monad funs → Functor/Applicative funs
* [ ] Generic types (e.g. `Monad m => m ()` where it really could just be a free `m a`)
* [ ] Imports

### CODE REVIEW

File: src/Aura/Packages/ABS.hs [hunk 1]
```diff
@@ -87,21 +87,21 @@ absDepsRepo = asks (getRepo . buildABSDeps)
 makeBuildable :: String -> String -> Aura Buildable
 makeBuildable repo name = do
   pb <- absPkgbuild repo name
-  return Buildable { baseNameOf   = name
+  pure Buildable { baseNameOf   = name
                    , pkgbuildOf   = pb
                    , isExplicit   = False
                    , buildScripts = \fp -> Just <$> copyTo repo name fp }
```
Comment (fosskers) [inline src/Aura/Packages/ABS.hs]: Fields here need to be aligned.

File: src/Bash/Simplify.hs [hunk 2]
```diff
@@ -60,31 +61,31 @@ simpNamespace :: Namespace -> Script -> Namespace
...
-replaceStr (Backtic f)   = Backtic `fmap` replace' f
+replaceStr (Backtic f)   = Backtic <$> replace' f
 
 expandStr :: [Either BashExpansion String] -> State Namespace [String]
 expandStr [] = pure []
 expandStr (x:xs) = do
   lhs <- expandStr' x
   rhs <- expandStr xs
-  pure $ lhs ++ rhs
+  pure $ lhs <> rhs
```
Comment (fosskers) [inline src/Bash/Simplify.hs]:
```haskell
expandStr (x:xs) = (<>) <$> expandStr' x <*> expandStr xs
```

**Comment by fosskers** [comment]
Poke :)

**Comment by bb010g** [comment]
Yeah. I'll push what I've got and then stick new stuff in a new PR.

**Comment by fosskers** [comment]
Thanks!

**Pull request merged**